### PR TITLE
2.0: remove redirection

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

File: etcd/client.go [hunk 1]
```diff
@@ -12,7 +12,10 @@ import (
 	"net/url"
 	"os"
 	"path"
+	"strings"
 	"time"
+
+	"github.com/coreos/etcd/etcdserver/etcdhttp/httptypes"
```
Comment (yichengq) [inline etcd/client.go]: use godep?

File: etcd/cluster.go [hunk 2]
```diff
@@ -18,34 +19,15 @@ func NewCluster(machines []string) *Cluster {
 
 	// default leader and machines
 	return &Cluster{
-		Leader:   machines[0],
+		Leader:   "",
 		Machines: machines,
+		picked:   rand.Intn(len(machines)),
```
Comment (yichengq) [inline etcd/cluster.go]: global rand needs to be seeded

File: etcd/cluster.go [hunk 3]
```diff
@@ -18,34 +19,15 @@ func NewCluster(machines []string) *Cluster {
...
-
-func (cl *Cluster) updateLeaderFromURL(u *url.URL) {
-	var leader string
-	if u.Scheme == "" {
-		leader = "http://" + u.Host
-	} else {
-		leader = u.Scheme + "://" + u.Host
-	}
-	cl.updateLeader(leader)
+	cl.Machines = strings.Split(machines, ",")
```
Comment (yichengq) [inline etcd/cluster.go]: it needs to update picked if the machines changes, and find a new picked if the old one is not in the list.

File: etcd/get.go [hunk 4]
```diff
@@ -18,9 +18,14 @@ func (c *Client) Get(key string, sort, recursive bool) (*Response, error) {
 }
 
 func (c *Client) RawGet(key string, sort, recursive bool) (*RawResponse, error) {
+	var q bool
+	if c.config.Consistency == STRONG_CONSISTENCY {
```
Comment (yichengq) [inline etcd/get.go]: do we have any way to set STRONG_CONSISTENCY now? i cannot find it anywhere.

Comment (yichengq) [inline etcd/get.go]: find it. We have public SetConsistency func

File: etcd/client.go (hunk 1 above)
Comment (xiang90) [inline etcd/client.go]: i do not want to bring in etcd's dependency into go-etcd, given the fact that we will move the client into etcd soon.

File: etcd/cluster.go (hunk 2 above)
Comment (xiang90) [inline etcd/cluster.go]: fixed

File: etcd/cluster.go (hunk 3 above)
Comment (xiang90) [inline etcd/cluster.go]: fixed

**Comment by yichengq** [comment]
lgtm. could you update the commit title?

**Pull request merged**